api/stats.proto

```diff
@@ -54,73 +54,84 @@ message StatsConfig {
-  // ``envoy.http_user_agent`` will be added with tag value ``ios``.
-  //
-  // The second regex will remove ``connection_manager_1.`` from the tag
-  // extracted name produced by the first regex
-  // ``http.connection_manager_1.user_agent.downstream_cx_total``, leaving
-  // ``http.user_agent.downstream_cx_total`` as the tag extracted name. The tag
-  // ``envoy.http_conn_manager_prefix`` will be added with the tag value
-  // ``connection_manager_1``.
-  string regex = 2;
+  oneof tag_value {
+    // Designates a tag to strip from the tag extracted name and provide as a named
+    // tag value for all statistics. This will only occur if any part of the name
+    // matches the regex provided with one or more capture groups.
+    //
+    // The first capture group identifies the portion of the name to remove. The
+    // second capture group (which will normally be nested inside the first) will
+    // designate the value of the tag for the statistic. If no second capture
+    // group is provided, the first will also be used to set the value of the tag.
+    // All other capture groups will be ignored.
+    //
+    // Take for example, with a stat name ``cluster.foo_cluster.upstream_rq_timeout``
+    // and
+    //
+    // .. code-block:: json
+    //
+    //   {
+    //     "tag_name": "envoy.cluster_name",
+    //     "regex": "^cluster\.((.+?)\.)"
+    //   }
+    //
+    // Note that the regex will remove ``foo_cluster.`` making the tag extracted
+    // name ``cluster.upstream_rq_timeout`` and the tag value for
+    // ``envoy.cluster_name`` will be ``foo_cluster`` (note: there will be no
+    // ``.`` character because of the second capture group).
+    //
+    // An example with two regexes and stat name
+    // ``http.connection_manager_1.user_agent.ios.downstream_cx_total``:
+    //
+    // .. code-block:: json
+    //
+    //   [
+    //     {
+    //       "tag_name": "envoy.http_user_agent",
+    //       "regex": "^http(?=\.).*?\.user_agent\.((.+?)\.)\w+?$"
+    //     },
+    //     {
+    //       "tag_name": "envoy.http_conn_manager_prefix",
+    //       "regex": "^http\.((.*?)\.)"
+    //     }
+    //   ]
+    //
+    // The first regex will remove ``ios.``, leaving the tag extracted name
+    // ``http.connection_manager_1.user_agent.downstream_cx_total``. The tag
+    // ``envoy.http_user_agent`` will be added with tag value ``ios``.
+    //
+    // The second regex will remove ``connection_manager_1.`` from the tag
+    // extracted name produced by the first regex
+    // ``http.connection_manager_1.user_agent.downstream_cx_total``, leaving
+    // ``http.user_agent.downstream_cx_total`` as the tag extracted name. The tag
+    // ``envoy.http_conn_manager_prefix`` will be added with the tag value
+    // ``connection_manager_1``.
+    string regex = 2;
+
+    // Add a default tag which have a fixed value and they will be added to all stats.
+    // Typical use case is identifying Envoy instances with the tag.
```

Comment:
The comment can be simplified as.

// Specifies a default tag value for the `tag_name`.

We don't need to care how the value is used, which can be controlled or stripped in different places.

Reply:
Just to add to that, can we remove the term `default` from the language since that implies it's a fallback value?  I think `fixed`, as you used below, is clearer.

Reply:
Agreed. Simplify the description and remove the term. Also I will remove the `default` term from Envoy side documents. ff78ff38cb2ab52dfe39338ec9ac9af0565393bb